projects/golang/math_big_fuzzer.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//go:build gofuzz
+// +build gofuzz
+
+package big
+
+import (
+	"strings"
+)
+
+func FuzzCmp(data []byte) int {
+	if !IsDivisibleBy(len(data), 2) {
+		return -1
+	}
+	half := len(data) / 2
+
+	halfOne := data[:half]
+	x, err := natFromString(string(halfOne))
+	if err != nil {
+		return 0
+	}
+
+	halfTwo := data[half:]
+	y, err := natFromString(string(halfTwo))
+	if err != nil {
+		return 0
+	}
+
+	x.cmp(y)
```

Comment:
General question: why fuzz the internal functions like `cmp` instead of converting the []byte to a `big.Int` using `big.Int.SetBytes` and then calling the exported `Cmp` function?

It would make the conversions from []byte easier, and ensure that the panics can be reachable by user code.

Reply:
Both fuzzers were rewritten to target exported functions. Does it look better?

Reply:
Yes it is looking better! Just a few other comments.